src/utils/date.ts

```diff
@@ -602,3 +602,21 @@ export const isValidDateString = (dateString: string) => {
   const parsedDate = parseISO(dateString);
   return isValid(parsedDate);
 };
+
+/**
+ * Recursively transforms iso date strings fields to Date fields.
+ */
+export const convertIsoStringFieldsToDate = (value: any): any => {
```

Comment:
It seems like we need to traverse all the structure to find this and might be a bit consuming after serialization, is better if this can be done during the JSON conversion or know the structure we are working to then convert the `time format` when required.

Reply:
Lets keep it in the back of our mind and think about optimising later if we find that it goes slow. Testing in simulator the mapping takes approximately 0.3ms per fare contract, and most phones are faster than the simulator. Also the mapping time is negligible when compared to the [slow rendering of long fare contract lists](https://github.com/AtB-AS/kundevendt/issues/19096).

Reply:
Great, this is nice to have in mind, might there be other places like this that all summed up might decrease performance.